confmap/confmap.go

```diff
@@ -189,6 +190,13 @@ func encoderConfig(rawVal interface{}) *encoder.EncoderConfig {
 	}
 }
 
+// case-sensitive version of the callback to be used in the MatchName property
+// of the DecoderConfig. The default for MatchEqual is to use strings.EqualFold,
+// which is case-insensitive.
+func caseSensitiveMatchName() func(mapKey, keyName string) bool {
+	return func(mapKey, keyName string) bool { return mapKey == keyName }
```

Comment:
```suggestion
func caseSensitiveMatchName() func(a, b string) bool {
	return func(a, b string) bool { return a == b }
```
nit: this is a generic function that works for any strings. There isn't that much value in using such long descriptive variable names that are only used on two lines.